### PR TITLE
Fix layout, try to generate JSON with LLM of places

### PR DESCRIPTION
**Pull request by sebazai**

### CODE REVIEW

File: src/app/api/places/chat/route.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    model: llama("openhermes2.5-mistral"),
+    system: `Your only task is to construct a valid Google Find Place API URL. 
+    You are allowed to use information provided in system prompt and the conversation.
+
+    Additional information:
+    ${userLocationString}
+    
+    Additional guidelines you should follow:
+      - The URL should be wrapped between three backticks
+      - Use radius of 2000 if it can't be determined from the conversation
```
Comment (mineralsfree) [inline src/app/api/places/chat/route.js:16]: unit is missing, 2000 meters I assume?

File: src/app/api/places/chat/route.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      - locationbias: A string specifying radius in meters, plus latitude/longitude in decimal degrees. Use the following format: circle:radius@latitude,longitude
+      - The Google API key is ${process.env.GOOGLE_API_KEY}
```
Comment (mineralsfree) [inline src/app/api/places/chat/route.js:18]: Why do we need to supply it a key here?

File: src/app/api/places/chat/route.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      - inputtype is textquery
+      - The fields you should include are: formatted_address, name, rating, opening_hours, geometry
+      - The input should be parsed from the conversation.
+
+    Example of a correctly generated URL:
+      - https://maps.googleapis.com/maps/api/place/findplacefromtext/json?fields=formatted_address%2Cname%2Crating%2Copening_hours%2Cgeometry&input=mongolian&inputtype=textquery&locationbias=circle:2000@47.6918452,-122.2226413&key=${process.env.GOOGLE_API_KEY}
```
Comment (mineralsfree) [inline src/app/api/places/chat/route.js:24]: We can insert key manually, there is no point of using unreliable AI to pass the key back and forth!

File: src/app/api/places/route.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import { createOllama } from "ollama-ai-provider";
+
+export async function POST(req) {
+  const llama = createOllama();
+  const request = await req.json();
+  const { prompt, data } = request;
+
+  const systemPrompt = `You are a helpful tour guide.
+
+    Your task is to find relevant information about places based on user input. Limit the search close to the user's location and to a maximum of five places.
```
Comment (mineralsfree) [inline src/app/api/places/route.js:12]:
```suggestion
    Your task is to find relevant places to visit based on user preferences expressed in user input . Limit the search close to the user's location and to a maximum of five places.
```

**Review by mineralsfree — APPROVED** [review]

File: src/app/api/places/chat/route.js (hunk 1 above)
Comment (sebazai) [reply src/app/api/places/chat/route.js:16]: The LLM will understand as docs talk about radius in numbers

File: src/app/api/places/chat/route.js (hunk 2 above)
Comment (sebazai) [reply src/app/api/places/chat/route.js:18]: To generate the URL. We won't supply the key if we use Cloud LLM's

**Pull request merged**